src/internal/got/client.ts

```diff
@@ -684,14 +686,41 @@ export class GotCFToolsClient implements CFToolsClient {
             return playerId;
         }
 
-        const response = await this.client.get<GetUserLookupResponse>('v1/users/lookup', {
-            searchParams: {
-                identifier: playerId.id,
-            },
-            context: {
-                authorization: await this.auth!.provide(this.client),
-            },
-        });
+        const requestUsesAccountCreation = this.hasAccountCreationAccess === true
+            && this.enterpriseToken !== undefined
+            && playerId instanceof SteamId64;
+
+        let response;
+        try {
+            response = await this.client.get<GetUserLookupResponse>('v1/users/lookup', {
+                searchParams: {
+                    identifier: playerId.id,
+                },
+                context: {
+                    authorization: await this.auth!.provide(this.client),
+                },
+            });
+        } catch (err) {
+            // Note: We could just apply the `create` param above, but this parameter is only ever
+            // allowed when there is no existing user with the supplied identity token.
+            if (requestUsesAccountCreation) {
+                response = await this.client.get<GetPlayerLookupResponseWithNotice>('v1/users/lookup', {
+                    searchParams: {
+                        identifier: playerId.id,
+                        create: true,
+                    },
+                    context: {
+                        authorization: await this.auth!.provide(this.client),
+                    },
+                });
+                if (response.notice !== 'Account Creation API account created') {
```

Comment:
Ehhh, this looks a bit fuzzy :D Do you have an example response of when the creation failed? Like is this really a 200 OK response?

Reply:
I do believe so yes, haha. I'll ask Phillip and get back to you about this!